src/main/java/dmytro/hadiuchko/springboot/service/impl/UserServiceImpl.java

```diff
@@ -30,6 +36,8 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto) {
         User user = userMapper.toModel(requestDto);
         user.setPassword(passwordEncoder.encode(user.getPassword()));
         user.setRoles(roleRepository.findAllByName(RoleName.ROLE_USER));
-        return userMapper.toUserResponse(repository.save(user));
+        User savedUser = repository.save(user);
+        shoppingCartRepository.save(shoppingCartService.createShoppingCart(savedUser));
+        return userMapper.toUserResponse(repository.save(savedUser));

```

Comment:
saving a savedUser does not look very good :)
```suggestion
        return userMapper.toUserResponse(savedUser);
```